src/relay/backend/compile_engine.cc

```diff
@@ -283,6 +305,248 @@ class ScheduleGetter :
   Array<Operation> scalars_;
 };
 
+// The getter to get shape function from functor.
```

Comment:
Can we possibly use a better name here? ShapeFuncGetter is not very descriptive imo, `MakeShapeFunc`, `GetShapeFunc`?